packages/material-ui-lab/src/TimelineDot/TimelineDot.js

```diff
@@ -1,67 +1,109 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
+import {
+  useThemeVariants,
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+} from '@material-ui/core/styles';
+import { deepmerge } from '@material-ui/utils';
 import { capitalize } from '@material-ui/core/utils';
-import { withStyles, useThemeVariants } from '@material-ui/core/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import timelineDotClasses, { getTimelineDotUtilityClass } from './timelineDotClasses';
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    display: 'flex',
-    alignSelf: 'baseline',
-    borderStyle: 'solid',
-    borderWidth: 2,
-    padding: 4,
-    borderRadius: '50%',
-    boxShadow: theme.shadows[1],
-    margin: '11.5px 0',
-  },
-  /* Styles applied to the root element if `variant="filled"`. */
-  filled: {},
-  /* Styles applied to the root element if `variant="outlined"`. */
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
 
-  outlined: {},
-  /* Styles applied to the root element if `color="grey"` and `variant="filled"`. */
-  filledGrey: {
-    borderColor: 'transparent',
-    color: theme.palette.grey[50],
-    backgroundColor: theme.palette.grey[400],
+  return deepmerge(
+    {
+      ...styles[
+        styleProps.color !== 'inherit' && `${styleProps.variant}${capitalize(styleProps.color)}`
+      ],
+      [`& .${timelineDotClasses.filled}`]: styles.filled,
+      [`& .${timelineDotClasses.outlined}`]: styles.outlined,
+    },
+    styles.root || {},
+  );
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { color, variant, classes } = styleProps;
+
+  const slots = {
+    root: ['root', color !== 'inherit' && `${variant}${capitalize(color)}`],
+    filled: ['filled'],
+    outlined: ['outlined'],
```

Comment:
This is wrong. This component has a single slot, root. To move above
```suggestion
```

Reply:
Done!